atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1308,97 +1287,40 @@ private CfDef getCfForTable(TableReference tableRef, byte[] rawMetadata) {
                 HistoryExtractor.SUPPLIER);
     }
 
-    public <T, U> ClosableIterator<RowResult<U>> getRangeWithPageCreator(
-            TableReference tableRef,
+    private <T, U> ClosableIterator<RowResult<U>> getRangeWithPageCreator(
+            final TableReference tableRef,
+            final RangeRequest rangeRequest,
+            final long timestamp,
+            final ConsistencyLevel consistency,
+            final Supplier<ResultsExtractor<T, U>> resultsExtractor) {
+        RowGetter rowGetter = new RowGetter(clientPool, queryRunner, consistency, tableRef);
+        ColumnGetter columnGetter = new DelegatingColumnGetter();
```

Comment:
Then we'd have to pass things like `clientPool` and `queryRunner` to `getRangeWithPageCreator`, which is ugly. 

Also, there will be a second ColumnGetter in #834.